### PR TITLE
fix: bugs in uts for polar and dipole fit

### PR DESCRIPTION
**Pull request by iProzd**
Fix following trivial bugs in dipole and polar fit uts:
1. `box` was not used in `extend_input_and_build_neighbor_list` (which means they were all tested in nopbc mode, if shifted coord is outside the box (sometimes) and normalized explicitly, results are not the same.) Input for fitting also used extended_atype instead of atype. (Only same when nopbc.)
2. Using of `mixed_types` is disordered, mismatched with descriptor or sometimes with nlist. Now only use `mixed_types`==False since the descriptor output is not in mixed types.

## Summary by CodeRabbit

- **Tests**
  - Improved consistency in parameter handling for various test methods.
  - Updated `mixed_types` parameter to dynamically use `self.dd0.mixed_types()` across multiple test functions for better flexibility and accuracy.

### CODE REVIEW

**Review by anyangml — APPROVED** [review]

File: source/tests/pt/model/test_dipole_fitting.py [hunk 1]
```diff
@@ -84,7 +83,7 @@ def test_consistency(
                 embedding_width=self.dd0.get_dim_emb(),
                 numb_fparam=nfp,
                 numb_aparam=nap,
-                mixed_types=mixed_types,
+                mixed_types=False,
```
Comment (wanghan-iapcm) [inline source/tests/pt/model/test_dipole_fitting.py:86]:
```suggestion
                mixed_types=dd0.mixed_type(),
```

File: source/tests/pt/model/test_dipole_fitting.py [hunk 2]
```diff
@@ -171,7 +169,7 @@ def test_rot(self):
...
```
Comment (wanghan-iapcm) [inline source/tests/pt/model/test_dipole_fitting.py:172]:
```suggestion
                mixed_types=self.dd0.mixed_type(),
```

File: source/tests/pt/model/test_dipole_fitting.py [hunk 3]
```diff
@@ -196,7 +194,7 @@ def test_rot(self):
                     _,
                     nlist,
                 ) = extend_input_and_build_neighbor_list(
-                    xyz + self.shift, atype, self.rcut, self.sel, not mixed_types
+                    xyz + self.shift, atype, self.rcut, self.sel, False, box=self.cell
```
Comment (wanghan-iapcm) [inline source/tests/pt/model/test_dipole_fitting.py:197]:
```suggestion
                    xyz + self.shift, atype, self.rcut, self.sel, self.dd0.mixed_type(), box=self.cell
```

File: source/tests/pt/model/test_dipole_fitting.py [hunk 4]
```diff
@@ -231,7 +229,7 @@ def test_permu(self):
                 _,
                 nlist,
             ) = extend_input_and_build_neighbor_list(
-                coord[idx_perm], atype, self.rcut, self.sel, True
+                coord[idx_perm], atype, self.rcut, self.sel, False, box=self.cell
```
Comment (wanghan-iapcm) [inline source/tests/pt/model/test_dipole_fitting.py:232]:
```suggestion
                coord[idx_perm], atype, self.rcut, self.sel, self.dd0.mixed_type(), box=self.cell
```

**Review by wanghan-iapcm — CHANGES_REQUESTED** [review]
better to use the `mixed_type` attribute of the descriptor. 

i have provided a few examples, please revise all the mixed_types parameters accordingly.

**Review by njzjz — APPROVED** [review]

**Review by wanghan-iapcm — APPROVED** [review]

**Pull request merged**